src/signer.rs

```diff
@@ -226,6 +227,25 @@ impl LightsparkSigner {
         Ok(self.build_commitment_secret(commitment_seed, per_commitment_point_idx))
     }
 
+    pub fn get_payment_preimage_nonce(&self) -> u64 {
+        let mut rng = OsRng;
+        rng.next_u64()
+    }
+
+    pub fn generate_preimage(&self, nonce: u64) -> Vec<u8> {
+        let key = self.derive_key("m/4h".to_owned()).unwrap();
```

Comment:
Does this ever return `Err`? If so, this should probably be a result rather than crash, right?

Reply:
Ah good catch! I'll add to the return value.